packages/base-driver/test/e2e/express/server.e2e.spec.js

```diff
@@ -1,16 +1,40 @@
-import {server} from '../../../lib';
+import {SslHandler, server} from '../../../lib';
 import axios from 'axios';
 import {createSandbox} from 'sinon';
 import B from 'bluebird';
 import _ from 'lodash';
 import {TEST_HOST, getTestPort} from '@appium/driver-test-support';
+import {given} from 'mocha-testdata';
+import {promisify} from 'util';
+
+const exec = promisify(require('child_process').exec);
+
+let givens = [{protocol: 'http'}, {protocol: 'https'}];
+
+async function startHttpsServer(port, routeConfiguringFunction, serverUpdaters = []) {
+  await exec('openssl genrsa 2048 > private.pem');
```

Comment:
where will this file be generated? should we not use an absolute path to be sure we're not littering test data around the filesystem or overwriting files?

Reply:
use a JS wrapper for generating a keypair.  this will fail when we add windows to the matrix.